assets/src/blocks/Submenu/SubmenuFrontend.js

```diff
@@ -1,20 +1,20 @@
-import { Fragment, useEffect } from '@wordpress/element';
-import { getSubmenuStyle, addSubmenuActions } from './submenuFunctions';
+import { getSubmenuStyle } from './getSubmenuStyle';
 import { SubmenuItems } from './SubmenuItems';
-import { useSubmenuItemsLoad } from './useSubmenuItemsLoad';
+import { makeHierarchical } from './makeHierarchical';
+import { getHeadingsFromDom } from './getHeadingsFromDom';
+import { BackTop } from './BackTop';
 
 export const SubmenuFrontend = ({ title, className, levels, submenu_style }) => {
 
-  const { menuItems } = useSubmenuItemsLoad(levels, false);
-
-  useEffect(() => addSubmenuActions(menuItems), [menuItems]);
-
+  const headings = getHeadingsFromDom(levels);
+  const menuItems = makeHierarchical(headings);
   const style = getSubmenuStyle(className, submenu_style);
 
   return (
-    <section className={`block submenu-block submenu-${style}`}>
-      <h2>{title}</h2>
-      <SubmenuItems menuItems={menuItems} />
+    <section className={ `block submenu-block submenu-${ style }` }>
+      <h2>{ title }</h2>
+      <SubmenuItems menuItems={ menuItems }/>
+      <BackTop/>
```

Comment:
Having the `BackTop` element here (inside the section) causes z-index issues and makes the button disappear under the footer when you scroll all the way to the bottom... That's why previously I appended it to the body instead!

Reply:
Yeah we should probably decouple the backtop from the component in some way, maybe we can make it a page level option? We could then turn on that option whenever the page contains a Submenu block

Reply:
For now I just included the back to top markup into master theme here https://github.com/greenpeace/planet4-master-theme/pull/1166/files . Still need to add something to toggle its visibility when you're at the top of the page. But already fixed the z-index issue.